pkg/config/types/gen_viper/generate.go

```diff
@@ -57,21 +57,20 @@ func generateSetDefaults(t reflect.Type, prefix string, path []string, value ref
 			// Inserting the viper.SetDefault for the nested struct path.
 			constantNameForStruct := strings.ReplaceAll(newPrefix, ".", "")
 			defaultValueForStruct := "cfg." + strings.Join(newPath, ".")
-			fmt.Fprintf(writer, "\tviper.%s(%s, %s)\n", methodName, constantNameForStruct, defaultValueForStruct)
+			fmt.Fprintf(writer, "\tp.Viper.%s(%s, %s)\n", methodName, constantNameForStruct, defaultValueForStruct)
```

Comment:
is that an accidental `\t -> \tp.` ?

Reply:
This is intentional: https://github.com/bacalhau-project/bacalhau/pull/3012/files#diff-5c467aa6d981a438438edace2b47a85fc4586b732c1f0af74bb10491c5db8d77
The code generated method now accepts an optional viper instance to set values on.